--- opencryptoki-2.4/configure.in.orig	2011-05-18 00:27:00.000000000 +0200
+++ opencryptoki-2.4/configure.in	2011-06-19 20:21:06.992291554 +0200
@@ -624,7 +624,7 @@
 
 CFLAGS="$CFLAGS -DPKCS64 -D_XOPEN_SOURCE=500"
 
-CFLAGS+=' -DCONFIG_PATH=\"$(localstatedir)/lib/opencryptoki\" -DSBIN_PATH=\"$(sbindir)\" -DLIB_PATH=\"$(libdir)\"'
+CFLAGS="$CFLAGS"' -DCONFIG_PATH=\"$(localstatedir)/lib/opencryptoki\" -DSBIN_PATH=\"$(sbindir)\" -DLIB_PATH=\"$(libdir)\"'
 
 # At this point, CFLAGS is set to something sensible
 AC_PROG_CC
